sycl/doc/extensions/proposed/sycl_ext_oneapi_launch_queries.asciidoc

```diff
@@ -125,135 +124,153 @@ class kernel {
-|Return Type
-|Description
-
-|`template <uint32_t Dimensions>
- max_work_item_sizes`
-|`sycl::queue`
-|`id<Dimensions>`
-|Returns the maximum number of work-items that are permitted in each dimension
- of a work-group, when the kernel is submitted to the specified queue,
- accounting for any kernel properties or features. If the kernel can be
- submitted to the specified queue without an error, the minimum value returned
- by this query is 1, otherwise it is 0.
-
-|`max_work_group_size`
-|`sycl::queue`
-|`size_t`
-|Returns the maximum number of work-items that are permitted in a work-group,
+_Returns_: The maximum number of work-items that are permitted in a work-group,
 when the kernel is submitted to the specified queue, accounting for any
-kernel properties or features. If the kernel can be submitted to the specified
-queue without an error, the minimum value returned by this query is 1,
-otherwise it is 0.
-
-|`max_num_work_groups`
-|`sycl::queue`, `sycl::range`, `size_t`
-|`size_t`
-|Returns the maximum number of work-groups, when the kernel is submitted to the
-specified queue with the specified work-group size and the specified amount of
-dynamic work-group local memory (in bytes), accounting for any kernel
-properties or features. If the specified work-group size is 0, which is
-invalid, then the implementation will throw a synchronous exception with the
-`errc::invalid` error code. If the kernel can be submitted to the specified
-queue without an error, the minimum value returned by this query is 1,
-otherwise it is 0.
+kernel properties or features.
+If the kernel can be submitted to the specified queue without an error, the
+minimum value returned by this query is 1, otherwise it is 0.
 
-|===
+'''
 
-A separate set of launch queries can be used to reason about how an
-implementation will launch a kernel on the specified queue. The values of these
-queries should also be checked if a kernel is expected to be launched in a
-specific way (e.g., if the kernel requires two sub-groups for correctness).
+[source,c++]
+----
+template <typename Param>
+size_t ext_oneapi_get_info(sycl::queue q, sycl::range<1> r, size_t bytes = 0) const;
+
+template <typename Param>
+size_t ext_oneapi_get_info(sycl::queue q, sycl::range<2> r, size_t bytes = 0) const;
 
-[source, c++]
+template <typename Param>
+size_t ext_oneapi_get_info(sycl::queue q, sycl::range<3> r, size_t bytes = 0) const;
 ----
-namespace ext::oneapi::experimental::info::kernel {
+_Constraints_: `Param` is `max_num_work_groups`.
 
-struct max_sub_group_size;
-struct num_sub_groups;
+_Returns_: The maximum number of work-groups, when the kernel is submitted to
+the specified queue with the specified work-group size and the specified amount
+of dynamic work-group local memory (in bytes), accounting for any kernel
+properties or features.
+If the specified work-group size is 0, which is invalid, then the
+implementation will throw a synchronous exception with the `errc::invalid`
```

Comment:
Are you saying that `ext_oneapi_get_info` will throw an exception in this case?  If so, use a *Throws:* paragraph instead:

> *Throws:* A synchronous `exception` with the error code `errc::invalid` if the work-group size `r` is 0.

Reply:
Good point, this should be a _Throws_ paragraph.  I think this was mixed in with the return value because the text came from one of the old-style tables.

I've fixed this in 3ccbfa7d821646ae5a41eb837c93227ab008029c, and I've also added the same exception to the other queries accepting a work-group size.  If we're going to throw an exception for a work-group size of 0, we should do it everywhere.